posts/2022-01-31-configuring-ocp-for-user-namespace.md

```diff
@@ -0,0 +1,198 @@
+  Jan 26 06:50:16 crc-hsl9k-master-0 bash[1658]: Processing packages...done
+  Jan 26 06:50:16 crc-hsl9k-master-0 bash[1658]: Running pre scripts...done
+  Jan 26 06:50:16 crc-hsl9k-master-0 bash[1658]: Running post scripts...done
+  Jan 26 06:50:17 crc-hsl9k-master-0 bash[1658]: Running posttrans scripts...done
+  Jan 26 06:50:17 crc-hsl9k-master-0 bash[1658]: Writing rpmdb...done
+  Jan 26 06:50:18 crc-hsl9k-master-0 bash[1658]: Writing OSTree commit...done
+  Jan 26 06:50:19 crc-hsl9k-master-0 bash[1658]: Staging deployment...done
+  Jan 26 06:50:20 crc-hsl9k-master-0 systemd[1]: Stopping Install custom runc...
+  Jan 26 06:50:20 crc-hsl9k-master-0 systemd[1]: install-runc.service: Succeeded.
+  Jan 26 06:50:20 crc-hsl9k-master-0 systemd[1]: Stopped Install custom runc.
+  Jan 26 06:50:20 crc-hsl9k-master-0 systemd[1]: install-runc.service: Consumed 94ms CPU time
+  -- Reboot --
+  Jan 26 06:51:10 crc-hsl9k-master-0 bash[1656]: runc-1.0.3-992.rhaos4.10.el8.x86_64
+  Jan 26 06:51:09 crc-hsl9k-master-0 systemd[1]: Started Install custom runc.
+  Jan 26 06:51:09 crc-hsl9k-master-0 systemd[1]: install-runc.service: Succeeded.
+  Jan 26 06:51:09 crc-hsl9k-master-0 systemd[1]: install-runc.service: Consumed 11ms CPU time
+  ```
+
+- For the cgroup2:
+
+  ```sh
+  ssh -i ~/.crc/machines/crc/id_ecdsa core@192.168.130.11 mount | grep cgroup2
+  ```
+
+- For the kernelarguments:
+
+  ```sh
+  ssh -i ~/.crc/machines/crc/id_ecdsa core@192.168.130.11 cat /proc/cmdline
+  BOOT_IMAGE=(hd0,gpt3)/ostree/rhcos-36fd944867b0e491991a65f6f3b7209c937fe3bd7cdbd855c7c5d5a7070ce570/vmlinuz-4.18.0-305.28.1.el8_4.x86_64 random.trust_cpu=on console=tty0 console=ttyS0,115200n8 ignition.platform.id=qemu ostree=/ostree/boot.1/rhcos/36fd944867b0e491991a65f6f3b7209c937fe3bd7cdbd855c7c5d5a7070ce570/0 root=UUID=91ba4914-fd2b-4a7c-b498-28585a80a40e rw rootflags=prjquota systemd.unified_cgroup_hierarchy=1 cgroup_no_v1=all psi=1
+  ```
+
+- For the subid configuration:
+
+  ```sh
+  ssh -i ~/.crc/machines/crc/id_ecdsa core@192.168.130.11 cat /etc/subuid /etc/subgid
+  ```
+
+  ```raw
+  core:100000:65536
+  containers:200000:268435456
+  core:100000:65536
+  containers:200000:268435456
+  ```
+
+- For the cri-o configuration:
+
+  ```sh
+  ssh -i ~/.crc/machines/crc/id_ecdsa core@192.168.130.11 cat /etc/crio/crio.conf.d/99-crio-userns.conf
+  ```
+
+  ```raw
+  # https://github.com/cri-o/cri-o/blob/main/docs/crio.conf.5.md#crioruntimeruntimes-table
+  [crio.runtime.runtimes.runc]
+  allowed_annotations=["io.kubernetes.cri-o.userns-mode"]
+  ```
+
+  Now we can use the annotation below for using user namespaces into our pods:
+
+  ```yaml
+  apiVersion: v1
+  kind: Pod
+  metadata:
+    annotations:
+      io.kubernetes.cri-o.userns-mode: "auto:size=65536"
+  ```
+
+## Wrap up
+
+With this configuration we can quickly set up our OCP cluster for quickly
+experiment and investigate with user namespaces.
```

Comment:
Grammar: *... our OCP cluster to quickly experiment with and investigate user namespace.*

Reply:
Thanks! updated